### DISCUSSION TITLE
Major issue that could lead to permanent data loss on PC

### DISCUSSION THREAD

**Discussion by rufflez** [rrirower/DP2SceneGraphWizard#35] [General]
This is a major programming issue that should have never been released publicly.  If you go through the wizard without adjusting any settings, you will wipe your C:\ drive.  When you get to the step where you select the folder to download the files to, the default is the C:\ drive.  The files need to go into an empty folder, if you didn't already have an empty folder, the software will delete everything to make the folder empty.

Why doesn't the software just create a blank folder?  Seems like an easy and very logical thing to do.

**Comment by ferdiworks** [comment]
@rufflez I have used this software from the beginning of its release and have once or twice selected a folder that was not empty and it would not allow me to use a folder that wasn't empty in a past older version. I had to select another folder! I never seen it try to delete a folder that had data in it. So I find it odd that something would try to delete any content in a folder that was not empty. So with what you just wrote I wanted to test a folder with content and data to see if it would try or just delete the content, I was able to select a folder with another channel package in it and it created the channel and did not delete the old channel but left a new channel package on my desktop and I was able to open the folder and the old channel package was in the folder!

**Reply by ferdiworks** [reply]
@rufflez OK, I just tried this again with another folder that had some older pics and text files in it, I backed the folder up just in case, I selected the folder, and I ran the software (current version)then clicked the download button and it created the new channel, and left it on the desktop and the folder I selected to download to was intact w the images and text file! nothing deleted!

**Reply by rufflez** [reply]
Well, I chose the Desktop (thank God I did!) and it wiped out my entire Desktop.  I was able to restore it since I had a backup from a few days earlier.  The program didn't work, either.  Channel Feed Manager refuses to save changes to older items and its really caused me headaches.  I've resulted into just editing the JSON file by hand, at least that way I know it will be correct.

**Comment by rrirower** [comment]
@rufflez Thanks for the feedback.  It's important to get positive and negative feedback.

The field on the form that asks for the download location is a prompt.  It does NOT represent what is actually the current default value of that field.  The user must either type a folder location or select one.  After selecting the folder, the user is warned if there are any files/folders in the selected folder.

![Capture](https://github.com/rrirower/DP2SceneGraphWizard/assets/34550407/02da70fe-fa2c-40eb-a976-84c6614e97a4)

The only way the entire drive will be deleted is if the user types "C:\", or, navigates to the root drive and selects it.  Either way, the user is **warned** before removing any files.  The C:\ drive is not deleted automatically.

Either way, I will look to incorporate a check for the C:\ drive as a safeguard against disaster.  Beyond that, the user bears some amount of responsibility for his choice given that he is warned before anything is deleted.

Now, onto Channel Feed Manager....have you reported any problems, or posted any issues?  What type of problems are you having?  Your comment is pretty vague.  There are plenty of CFM users who are using that tool successfully.  I'd be glad to entertain your concerns.  Simply post in the CFM discussions or issues.

**Reply by rufflez** [reply]
So the issues I have with Channel Feed Manager is that its not reliable.  We switched hosting platforms around the same time JSON Feed Manager changed to Channel Feed Manager.  The UI has become much more of a hassle than the previous program.  One issue that I've had on both is that the UI doesn't show up and I have to uninstall the program and reinstall it for it to load.  

Issues I've encountered with Channel Feed Manager:

-  updated every item in my feed to reflect a new server we are hosting 
- ensured there were tags and genres associated with every video
- made sure my live feed has a thumbnail

Some of the videos have the old server for the thumbnails and video links, half the tags and genres haven't been applied, and I get an error everytime I try to update the live feed that there isn't a thumbnail when I'm staring right at the thumbnail url.  JSON Feed Manager worked better, didn't cause any data loss issues and only had 1 bug that never got fixed, but had an easy work around (if a video was > 1 hour it added 1 to the minutes from the wizard to the main UI).

**Reply by rrirower** [reply]
Andrew, thanks for continuing this discussion.  Can I ask what CFM version you are using?  Can you provide a sample feed file that demonstrates your issues?

**Reply by rufflez** [reply]
Sure: Version 1.0.0.6
link to feed:
https://hmsepisodes.s3.us-east-2.amazonaws.com/roku_json/roku_channel.json

I think the issue stems from which item I update.  In JSON Feed Manager, there was a single list.  Now, there is a default list, and each category has a list.  I've been updating the default, and its not saving (or the other way around).  Honestly, if I could just get JSON feed manager exe, I would go back to that in an instant.  But Channel Feed Manager is not usable for me.  I'd rather just create the JSON code myself, which is what I've been doing for our SDK conversion.

Also: If you've updated an item like tags, genres there is no feedback in the system that you've done it.  You can select an item in the json that has a tag or genre in there, go back into channel feed manager and select that item, and they are blanked out.

**Reply by rrirower** [reply]
I suspect that you've uncovered a few unreported bugs.  I've fixed the thumbnail issue which affected liveFeed objects only.  And, I believe the other issues are related to your exclusive use of playlists as Categories.  Most users use a mixture of Categories based on queries and playlists.  I think that's why you're seeing the 'Default' category.  'Default' is used to bucket content that is not assigned to a Category.  This is most likely another bug I should be able to fix.

Most of your frustration could have been avoided if you had reported these problems when they were encountered.

**Reply by rrirower** [reply]
I've fixed the 'Default' assignment issue.  It was actually a pretty easy bug to fix.

![Capture](https://github.com/rrirower/DP2SceneGraphWizard/assets/34550407/92612907-bcd3-4979-b451-07c3c57e3bca)

In debugging, I found that your feed file defines Categories based on Playlists for all but one item of content.  However, your feed content also defines a tag for those items that are part of a Playlist.  Defining a tag implies that the feed uses Categories based on a query rather than a Playlist.  You may want to clean that up.

**Reply by rrirower** [reply]
@rufflez Your issues should be resolved in the next release.

**Comment by rrirower** [comment]
A new version 1.0.0.7 was released to resolve the CFM issues.

**Comment by adamesq** [comment]
WOW...I see it - just 12 minutes ago! AWESOME.

While launch is pending, I just noticed the Package Utility screenshot button. It's telling me:

HD mode 1280x720 image required for channel store upload
Screenshot not ok

I don't *think* this will cause a publishing fail...do you know?

**Reply by ferdiworks** [reply]
@adamesq  No you will not get a publishing fail for the screenshot!

**Reply by adamesq** [reply]
Thank you, @ferdiworks - just confirmed first hand...I got the successfully published notification a few minutes ago!!!👏🏻👏🏻👏🏻 It's been a very, very long few days stressing about this DP to SDK move!
